src/main/java/org/jenkinsci/plugins/docker/commons/fingerprint/DockerFingerprints.java

```diff
@@ -254,55 +254,57 @@ public static void addRunFacet(@Nonnull ContainerRecord record, @Nonnull Run<?,?
+                try {
+                    if (descendantFacet == null) {
+                        descendantFacet = new DockerDescendantFingerprintFacet(f, timestamp, ancestorImageId);
+                        facets.add(descendantFacet);
+                    }
+                    descendantFacet.addDescendantImageId(descendantImageId);
+                    descendantFacet.addFor(run);
+                    DockerFingerprintAction.addToRun(f, ancestorImageId, run);
+                    bc.commit();
+                } finally {
+                    bc.abort();
+                }
+            }
+            Fingerprint f = forImage(run, descendantImageId);
             Collection<FingerprintFacet> facets = f.getFacets();
-            DockerDescendantFingerprintFacet descendantFacet = null;
+            DockerAncestorFingerprintFacet ancestorFacet = null;
             for (FingerprintFacet facet : facets) {
-                if (facet instanceof DockerDescendantFingerprintFacet) {
-                    descendantFacet = (DockerDescendantFingerprintFacet) facet;
+                if (facet instanceof DockerAncestorFingerprintFacet) {
+                    ancestorFacet = (DockerAncestorFingerprintFacet) facet;
                     break;
                 }
             }
             BulkChange bc = new BulkChange(f);
             try {
-                if (descendantFacet == null) {
-                    descendantFacet = new DockerDescendantFingerprintFacet(f, timestamp, ancestorImageId);
-                    facets.add(descendantFacet);
+                if (ancestorFacet == null) {
+                    ancestorFacet = new DockerAncestorFingerprintFacet(f, timestamp, descendantImageId);
+                    facets.add(ancestorFacet);
+                }
+                if (ancestorImageId != null) {
+                    ancestorFacet.addAncestorImageId(ancestorImageId);
                 }
-                descendantFacet.addDescendantImageId(descendantImageId);
-                descendantFacet.addFor(run);
-                DockerFingerprintAction.addToRun(f, ancestorImageId, run);
+                ancestorFacet.addFor(run);
+                DockerFingerprintAction.addToRun(f, descendantImageId, run);
                 bc.commit();
             } finally {
                 bc.abort();
             }
         }
-        Fingerprint f = forImage(run, descendantImageId);
-        Collection<FingerprintFacet> facets = f.getFacets();
-        DockerAncestorFingerprintFacet ancestorFacet = null;
-        for (FingerprintFacet facet : facets) {
-            if (facet instanceof DockerAncestorFingerprintFacet) {
-                ancestorFacet = (DockerAncestorFingerprintFacet) facet;
-                break;
-            }
-        }
-        BulkChange bc = new BulkChange(f);
-        try {
-            if (ancestorFacet == null) {
-                ancestorFacet = new DockerAncestorFingerprintFacet(f, timestamp, descendantImageId);
```

Comment:
Unreadable diff. [This](https://github.com/jenkinsci/docker-commons-plugin/pull/69/files?w=1) is much easier to follow, but generally I recommend not changing indentation of large code blocks, since it messes up other things like cherry-picks and merges.